spec/services/retrieve_pipeline_viz_graph_data_service_spec.rb

```diff
@@ -109,9 +109,9 @@
     before do
       # Project and sample required to create pipeline run.
       project = create(:public_project)
-      sample = create(:sample, project: project)
+      @sample = create(:sample, project: project)
```

Comment:
Why do you need the `@` ?

Reply:
In order to create a new pipeline run with modified stage data in the new test, I need to provide a sample as the argument; the @ makes the sample available for the entire instance so that I can use it in the new test below